src/java.base/share/classes/java/net/URLEncoder.java

```diff
@@ -239,11 +241,16 @@ public static String encode(String s, Charset charset) {
         }
 
         StringBuilder out = new StringBuilder(s.length() << 1);
-        CharArrayWriter charArrayWriter = new CharArrayWriter();
         if (i > 0) {
             out.append(s, 0, i);
         }
 
+        CharsetEncoder ce = charset.newEncoder()
+                .onMalformedInput(CodingErrorAction.REPLACE)
+                .onUnmappableCharacter(CodingErrorAction.REPLACE);
```

Comment:
@dfuch The explicit use of the REPLACE action makes me wonder if the spec should be clarified at some point as it doesn't seem to be documented.

Reply:
> I'd be happy to spell it out in the documentation, just pointing out that it's been long-standing behavior.

Right, I wasn't suggesting there is behavior change, instead I was pointing out to Daniel that the spec is incomplete and we should clarify it, it would be a separate update/PR and would require a CSR to track the spec update.

Reply:
@AlanBateman I have logged [JDK-8316734](https://bugs.openjdk.org/browse/JDK-8316734)